### ISSUE TITLE
`Style/Documentation` macro-only check is incorrect

### ISSUE THREAD

**Issue by ghiculescu** [rubocop/rubocop#11293]
This does not get a lint error:

```ruby
class Thing < ApplicationRecord
  belongs_to :user
end
```

I think it should, because it is a class with legitimate contents, but no documentation.

I think the changes made in https://github.com/rubocop/rubocop/pull/8812 have introduced a false positive.

--------

## Expected behavior

`Style/Documentation` lint error, as the class has no documentation.

## Actual behavior

No lint error.

## RuboCop version

```
1.40.0 (using Parser 3.1.3.0, rubocop-ast 1.24.0, running on ruby 3.1.3) [x86_64-darwin20]
  - rubocop-minitest 0.25.0
  - rubocop-performance 1.15.1
  - rubocop-rails 2.17.3
  - rubocop-thread_safety 0.4.4
```
